### PR TITLE
Correct the indexes on user_favourites table

### PR DESCRIPTION
**Pull request by bharat619**
As in the title

### CODE REVIEW

File: app/models/user_favourite.rb [hunk 1]
```diff
@@ -14,35 +14,36 @@ class UserFavourite < ApplicationRecord
...
     # Mark a record as a favourite
     #
     def add_user_favourite(record, persistent: false, user: User.current_user)
       return unless user.present?
-      
+
       UserFavourite.where(
         user: user,
         favourite: record,
         persistent: persistent
```
Comment (patrixr) [inline app/models/user_favourite.rb:39]: gotta remove this line here

**Review by patrixr — CHANGES_REQUESTED** [review]
Can you add a test for this as well ?

- create a persistent favourite
- create a non-persistent favourite for the same record and user
- should not try to create a new record

File: app/models/user_favourite.rb (hunk 1 above)
Comment (patrixr) [reply app/models/user_favourite.rb:39]:
```suggestion
```

File: app/models/user_favourite.rb [hunk 2]
```diff
@@ -14,35 +14,36 @@ class UserFavourite < ApplicationRecord
...
       ).first_or_initialize do |fav|
+        fav.persistent = fav.persistent ? true : persistent
```
Comment (swatijadhav) [inline app/models/user_favourite.rb:41]:
```suggestion
        fav.persistent = fav.persistent || persistent
```

**Review by patrixr — APPROVED** [review]

**Pull request merged**